### PR TITLE
Add lookup table API

### PR DESCRIPTION
**Pull request by emord**
https://dimagi-dev.atlassian.net/browse/HI-582

https://confluence.dimagi.com/display/commcarepublic/Lookup+Table+API

This is going to introduce a lookup table API (similar to our other APIs) so large projects like the one in the interrupt ticket don't have a huge excel file upload with little errors on what failed and when (I'm going to have another PR documenting how terrible CouchTransaction is and why it should never be used)

I started this with TDD , but couldn't actually run the tests due to some accounting error which was sad. So I'm openeing for tests (EDIT: I was able to figure out the testing issues)

TODO:

- [x] decide on what to do with item_attributes (and what exactly they are/do)
- [x] Resolve the test changes with https://github.com/dimagi/commcare-hq/commit/ea1a43dce91223f90104dc4eb6c62b0053ce36f8#diff-369d68777342db985fd1b4269bf45b7d
- [x] actually implement sort_key in object creation
- [x] add comments about which python methods relate to HTTP methods
- [x] after deploy, write the appropriate documentation for confluence for this API

### CODE REVIEW

**Comment by emord** [comment]
This is ready for review. This is generally just opening up the ability to create/delete/update lookup tables and their rows via the API one at a time instead of needing to create their tables all at once and running into https://github.com/dimagi/commcare-hq/pull/24229/files

File: corehq/apps/api/tests/lookup_table_resources.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            "tag": "lookup_table"
+        }
+
+    def test_get_list(self):
+        response = self._assert_auth_get_resource(self.list_endpoint)
+        self.assertEqual(response.status_code, 200)
+
+        fixture_data_types = json.loads(response.content)['objects']
+        self.assertEqual(len(fixture_data_types), 1)
+        self.assertEqual(fixture_data_types, [self._data_type_json()])
```
Comment (orangejenny) [inline corehq/apps/api/tests/lookup_table_resources.py]: Why doesn't this need to be `assertListEqual`? (Same for `TestLookupTableItemResource .test_get_list`)

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
Kicked build. `test_duplicate_client_claim` strikes again.

File: corehq/apps/api/tests/lookup_table_resources.py (hunk 1 above)
Comment (emord) [reply corehq/apps/api/tests/lookup_table_resources.py]: `assertEqual` gets mapped to the type appropriate equality comparison https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual (and you can add to that if you're subclassing `TestCase` / `SimpleTestCase`)

Comment (orangejenny) [reply corehq/apps/api/tests/lookup_table_resources.py]: did not know that (obviously), thanks

**Pull request merged**